### PR TITLE
Provides all local data sources

### PR DESCRIPTION
**Pull request by Turhsus**
PR for all data sources. This thing is huge, will take a little time to get through, a lot of the code is the same tho.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+## 6.31.0 (February 25, 2023).
```
Comment (alexhung) [inline CHANGELOG.md:1]: 25th is a Saturday 😄 Let's change to 27th and gives @danielmkn time to review as well.

File: docs/data-sources/repository/local/alpine_repository.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+---
+
+# Artifactory Local Alpine Repository Data Source
+
+Retrieves a local alpine repository.
+
+## Example Usage
+
+```hcl
+data "artifactory_local_alpine_repository" "terraform-local-test-alpine-repo-basic" {
```
Comment (alexhung) [inline docs/data-sources/repository/local/alpine_repository.md:12]: I'd remove the `terraform-` prefix here (and from the others too).

File: docs/data-sources/repository/local/cargo_repository.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+The following argument is supported:
+
+* `key` - (Required) the identity key of the repo.
+
+## Attribute Reference
+
+The following attributes are supported, along with the [common list of attributes for the local repositories](local.md):
+
+* `anonymous_access` - (Optional) Cargo client does not send credentials when performing download and search for crates.
```
Comment (alexhung) [inline docs/data-sources/repository/local/cargo_repository.md:27]: I merged https://github.com/jfrog/terraform-provider-artifactory/pull/661 yesterday so you should include the new attribute.

File: docs/data-sources/repository/local/debian_repository.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+---
+
+# Artifactory Local Debian Repository Data Source
+
+Retrieves a local Debian repository.
+
+## Example Usage
+
+```hcl
+data "artifactory_local_debian_repository" "my-debian-repo" {
```
Comment (alexhung) [inline docs/data-sources/repository/local/debian_repository.md:12]: Same here with the `my-` prefix. Let's be consistent among our own documentation. I have no strong opinion on what prefix we want to use. `my-` is fine. I just want consistency.

File: docs/data-sources/repository/local/ivy_repository.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+The following argument is supported:
+
+* `key` - (Required) the identity key of the repo.
+
+## Attribute Reference
+
+There are no unique attributes for , along with the [common list of attributes for the local repositories](local.md):
+
+[//]: # (Are these attributes supported for this repo type?)
```
Comment (alexhung) [inline docs/data-sources/repository/local/ivy_repository.md:27]: ?

File: docs/data-sources/repository/local/sbt_repository.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+The following argument is supported:
+
+* `key` - (Required) the identity key of the repo.
+
+## Attribute Reference
+
+There are no unique attributes for , along with the [common list of attributes for the local repositories](local.md):
+
+[//]: # (Are these attributes supported for this repo type?)
```
Comment (alexhung) [inline docs/data-sources/repository/local/sbt_repository.md:27]: ?

File: pkg/artifactory/datasource/repository/local/datasource_artifactory_local_docker_repository.go [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+	return &schema.Resource{
+		Schema:      local.DockerV2LocalSchema,
+		ReadContext: MkRepoReadDataSource(pkr, constructor),
+	}
+}
+
+func DataSourceArtifactoryLocalDockerV1Repository() *schema.Resource {
+	// this is necessary because of the pointers
+	skeema := util.MergeMaps(local.DockerV1LocalSchema)
```
Comment (alexhung) [inline pkg/artifactory/datasource/repository/local/datasource_artifactory_local_docker_repository.go:36]: I wonder why this is necessary for this but for not the others?

File: pkg/artifactory/datasource/repository/local/datasource_artifactory_local_nuget_repository.go [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+		}, nil
+	}
+
+	return &schema.Resource{
+		Schema:      local.NugetLocalSchema,
+		ReadContext: MkRepoReadDataSource(packer.Default(local.NugetLocalSchema), constructor),
+		Description: "Data Source for a local nuget repository",
+	}
+
+	//return repository.MkResourceSchema(NugetLocalSchema, packer.Default(NugetLocalSchema), unPackLocalNugetRepository, constructor)
```
Comment (alexhung) [inline pkg/artifactory/datasource/repository/local/datasource_artifactory_local_nuget_repository.go:27]: Please clean up

File: pkg/artifactory/datasource/repository/local/datasource_artifactory_local_repository_test.go [hunk 9]
```diff
@@ -63,3 +65,682 @@ func mkTestCase(repoType string, t *testing.T) (*testing.T, resource.TestCase) {
...
+		"anonymous_access": test.RandBool(),
+		"name":             name,
+	}
+	localRepositoryBasic := util.ExecuteTemplate("TestAccLocalCargoRepository", `
+		resource "artifactory_local_cargo_repository" "{{ .name }}" {
+		  key                 = "{{ .name }}"
+		  anonymous_access = {{ .anonymous_access }}
+		}
+
+    data "artifactory_local_cargo_repository" "{{ .name }}" {
```
Comment (alexhung) [inline pkg/artifactory/datasource/repository/local/datasource_artifactory_local_repository_test.go:168]: Let's have consistent indentation *within* the template 😄 (this happens a number of times in this file).

**Review by alexhung — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Turhsus) [reply CHANGELOG.md:1]: yea that was the goal I just got my dates way off XD

File: docs/data-sources/repository/local/cargo_repository.md (hunk 3 above)
Comment (Turhsus) [reply docs/data-sources/repository/local/cargo_repository.md:27]: cdnRedirect right? will do, will do a merge commit from master then go from there

File: docs/data-sources/repository/local/debian_repository.md (hunk 4 above)
Comment (Turhsus) [reply docs/data-sources/repository/local/debian_repository.md:12]: Sounds good, no my

File: pkg/artifactory/datasource/repository/local/datasource_artifactory_local_repository_test.go (hunk 9 above)
Comment (Turhsus) [reply pkg/artifactory/datasource/repository/local/datasource_artifactory_local_repository_test.go:168]: Damn thought I made consistent the intellij indentation settings, guess I didn't fix it. Will check again

File: docs/data-sources/repository/local/cargo_repository.md (hunk 3 above)
Comment (alexhung) [reply docs/data-sources/repository/local/cargo_repository.md:27]: See the PR. It's the Cargo's `enabled_sparse_index` attribute.

File: pkg/artifactory/datasource/repository/local/datasource_artifactory_local_docker_repository.go (hunk 7 above)
Comment (Turhsus) [reply pkg/artifactory/datasource/repository/local/datasource_artifactory_local_docker_repository.go:36]: I'm not sure, its done in the regular resource so I copied it over here. I've run into this pointer issue before, and it didn't work without this for a more complex test in the resource files. I will see if its necessary for data sources, kind of assumed it was

File: docs/data-sources/repository/local/sbt_repository.md (hunk 6 above)
Comment (Turhsus) [reply docs/data-sources/repository/local/sbt_repository.md:27]: There is a list of attributes here for an sbt repository that is copied and pasted from gradle repository list. I'm not sure if an sbt repository has those attributes though. The data structure that its built on does, but the resource file doesn't list those attributes.

Comment (alexhung) [reply docs/data-sources/repository/local/sbt_repository.md:27]: Ok. Regardless of what we end up putting in this doc file, what we have here is not correct.

File: pkg/artifactory/datasource/repository/local/datasource_artifactory_local_nuget_repository.go [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+	"github.com/jfrog/terraform-provider-artifactory/v6/pkg/artifactory/resource/repository/local"
+	"github.com/jfrog/terraform-provider-shared/packer"
+)
+
+func DataSourceArtifactoryLocalNugetRepository() *schema.Resource {
+	constructor := func() (interface{}, error) {
+		return &local.NugetLocalRepositoryParams{
+			RepositoryBaseParams: local.RepositoryBaseParams{
+				PackageType: "nuget",
+				Rclass:      "local",
```
Comment (alexhung) [inline pkg/artifactory/datasource/repository/local/datasource_artifactory_local_nuget_repository.go:14]: Let's use `rclass` here that is defined already in `local.go`

**Review by alexhung — APPROVED** [review]
LGTM :shipit:

**Review by danielmkn — APPROVED** [review]

File: pkg/artifactory/datasource/repository/local/datasource_artifactory_local_nuget_repository.go (hunk 10 above)
Comment (danielmkn) [reply pkg/artifactory/datasource/repository/local/datasource_artifactory_local_nuget_repository.go:14]: I assume we're merging it as is, without this change?

Comment (alexhung) [reply pkg/artifactory/datasource/repository/local/datasource_artifactory_local_nuget_repository.go:14]: Yea, I can update this in my PR.

Comment (Turhsus) [reply pkg/artifactory/datasource/repository/local/datasource_artifactory_local_nuget_repository.go:14]: Okay sounds good, i'll merge

**Pull request merged**